rpc/core/health.go

```diff
@@ -0,0 +1,51 @@
+package core
+
+import (
+	"time"
+
+	"errors"
+	"fmt"
+	ctypes "github.com/tendermint/tendermint/rpc/core/types"
+	"github.com/tendermint/tendermint/types"
+)
+
+// Get node health. Checks whether new blocks are created.
+//
+// ```shell
+// curl 'localhost:46657/health'
+// ```
+//
+// ```go
+// client := client.NewHTTP("tcp://0.0.0.0:46657", "/websocket")
+// result, err := client.Health()
+// ```
+//
+// > The above command returns JSON structured like this:
+//
+// ```json
+// {
+// 	"error": "",
+// 	"result": {},
+// 	"id": "",
+// 	"jsonrpc": "2.0"
+// }
+// ```
+func Health() (*ctypes.ResultHealth, error) {
+	var latestBlockMeta *types.BlockMeta
+	latestHeight := blockStore.Height()
+	if latestHeight != 0 {
+		latestBlockMeta = blockStore.LoadBlockMeta(latestHeight)
+	}
+
+	latestBlockTimeout := cfg.Consensus.Commit(latestBlockMeta.Header.Time).UnixNano()
+	currentTime := time.Now().UnixNano()
+	blockDelayTime := time.Duration(latestBlockTimeout-currentTime) * time.Nanosecond
+
+	// use single commit timeout as threshold for block creation
+	threshold := time.Duration(cfg.Consensus.TimeoutCommit) * time.Millisecond
+	if blockDelayTime > threshold {
```

Comment:
It's a bit more complex than that. If `no_empty_blocks` set to `true`, then this endpoint will report `Unhealthy` state, which is wrong.

I think we should for now just return `return &ctypes.ResultHealth{}, nil`

Later, we can implement @srmo request based on other peers heights (peerState) and this node height (blockStoreHeight). For that we'll need to add config variable (`UnhealthyHeightLag`)

Reply:
@melekes , remove whole check at all? 
I am a bit confused about, how we will know that blocks are been created and node is healthy. So, just return `return &ctypes.ResultHealth{}, nil`, and this will show that node hasn't crashed and still running ?

Reply:
Removed

Reply:
yeah, 200 OK response should inform a client that a node is running at the very least.